File: src/test/java/ukalus/persistence/Setter.java
/*
 * Created on May 18, 2004
 *  
 */
package ukalus.persistence;

import java.util.function.Function;

import java.io.Serializable;

/**
 * @author gremopm
 * 
 */
public class Setter<T extends Serializable> implements Function<Reference<T>, Object> {

  private static final long serialVersionUID = 3256442491011217201L;
  private T value = null;

  public Setter(T value) {
    this.value = value;
  }

  public Object apply(Reference<T> object) {
    object.set(value);
    return value;
  }

}

File: src/test/kotlin/ukalus/persistence/Setter.kt
/*
 * Created on May 18, 2004
 *  
 */
package ukalus.persistence

import java.util.function.Function

import java.io.Serializable

/**
 * @author gremopm
 */
class Setter<T>(private var value: T) : Function<Reference<T>, Any?>, Serializable {
    override fun apply(`object`: Reference<T>): Any? {
        `object`.set(value)
        return value
    }
}